dali/kernels/tensor_test.cc

```diff
@@ -0,0 +1,355 @@
+  TensorShape<DynamicDimensions> b(test_shape_vec);
+  ASSERT_EQ(b.size(), test_shape_vec.size());
+  for (int i = 0; i < test_shape_vec.size(); i++) {
+    ASSERT_EQ(b[i], test_shape_vec[i]);
+  }
+
+  // Expanded arguments constructor
+  TensorShape<DynamicDimensions> c(1);
+  ASSERT_EQ(c.size(), 1);
+  ASSERT_EQ(c[0], 1);
+
+  TensorShape<DynamicDimensions> d(1, 2, 3, 4);
+  ASSERT_EQ(d.size(), 4);
+  ASSERT_EQ(d[0], 1);
+  ASSERT_EQ(d[1], 2);
+  ASSERT_EQ(d[2], 3);
+  ASSERT_EQ(d[3], 4);
+
+  // Copy constructor
+  TensorShape<DynamicDimensions> check_construct(a);
+  for (int i = 0; i < test_dim; i++) {
+    ASSERT_EQ(check_construct[i], a[i]);
+  }
+
+  // Asignement
+  TensorShape<DynamicDimensions> check_assign;
+  check_assign = a;
+  ASSERT_EQ(check_assign.size(), a.size());
+  for (int i = 0; i < a.size(); i++) {
+    ASSERT_EQ(check_assign[i], a[i]);
+  }
+
+  // Second asignement to the same value
+  check_assign = b;
+  ASSERT_EQ(check_assign.size(), b.size());
+  for (int i = 0; i < b.size(); i++) {
+    ASSERT_EQ(check_assign[i], b[i]);
+  }
+
+    // Move rvalue
+  TensorShape<DynamicDimensions> check_move_construct(TensorShape<DynamicDimensions>{test_shape_arr});
+  for (int i = 0; i < test_dim; i++) {
+    ASSERT_EQ(check_move_construct[i], test_shape_arr[i]);
+  }
+
+  // Assignement for rvalue
+  TensorShape<DynamicDimensions> check_move_assign;
+  check_move_assign = TensorShape<DynamicDimensions>{test_shape_arr};
+  for (int i = 0; i < test_dim; i++) {
+    ASSERT_EQ(check_move_assign[i], test_shape_arr[i]);
+  }
+}
+
+
+TEST(TensorShapeTest, StaticDynamicConversions) {
+  TensorShape<3> static_shape_3(1, 2, 3);
+  TensorShape<5> static_shape_5(1, 2, 3, 4, 5);
+
+  TensorShape<DynamicDimensions> check_construct_3(static_shape_3);
+  ASSERT_EQ(check_construct_3.size(), static_shape_3.size());
+  for (int i = 0; i < static_shape_3.size(); i++) {
+    ASSERT_EQ(check_construct_3[i], static_shape_3[i]);
+  }
+
+  TensorShape<DynamicDimensions> check_construct_5(static_shape_5);
+  ASSERT_EQ(check_construct_5.size(), static_shape_5.size());
+  for (int i = 0; i < static_shape_5.size(); i++) {
+    ASSERT_EQ(check_construct_5[i], static_shape_5[i]);
+  }
+
+  TensorShape<DynamicDimensions> check_assign;
+  check_assign = static_shape_3;
+  ASSERT_EQ(check_assign.size(), static_shape_3.size());
+  for (int i = 0; i < static_shape_3.size(); i++) {
+    ASSERT_EQ(check_assign[i], static_shape_3[i]);
+  }
+
+  check_assign = static_shape_5;
+  ASSERT_EQ(check_assign.size(), static_shape_5.size());
+  for (int i = 0; i < static_shape_5.size(); i++) {
+    ASSERT_EQ(check_assign[i], static_shape_5[i]);
+  }
+
+  auto s3 = TensorShape<3>{2, 4, 6};
+  check_assign = s3;
+  static_shape_3 = check_assign.to_static_ndim<3>();
+  for (int i = 0; i < s3.size(); i++) {
+    ASSERT_EQ(static_shape_3[i], s3[i]);
+  }
+}
+
+TEST(TensorShapeTest, Comparisons) {
+  // Static ndim
+  ASSERT_EQ(TensorShape<1>(1) == TensorShape<1>(1), true);
```

Comment:
Use ASSERT_TRUE && ASSERT_FALSE. Or perhaps even try ASSERT_EQ and ASSERT_NE for the actual objects.

Reply:
I'am using them (ASSERT_EQ) most of the time but here I want to test the actual operators `==` and `!=` so I don't like to rely on internal implementation of gtest calling them.

Done.